core/src/main/java/io/opentelemetry/android/instrumentation/AndroidInstrumentationServices.kt

```diff
@@ -5,17 +5,19 @@
 
 package io.opentelemetry.android.instrumentation
 
+import io.opentelemetry.android.internal.instrumentation.AndroidInstrumentationServicesImpl
+
 /**
- * Stores and provides all the available instrumentations.
+ * Loads and provides Java SPI services of type [AndroidInstrumentation] from the runtime classpath.
  */
-interface AndroidInstrumentationRegistry {
+interface AndroidInstrumentationServices {
```

Comment:
Sorry to say so, but I don't love this name. It's leaking out the fact that instrumentations are discovered via `AutoService` (hence "Services") and suggests that the things you can `get()` from it are services (but they're not, they're instrumentation). We also have the `ServiceManager`, so putting `Service` in the name implies a relationship there, where we might not want one.

Some alternative names:
* `AndroidInstrumentionDiscovery`
* `DetectedInstrumentationLoader`
* `InstrumentationFinder`
* `AndroidInstrumentationLoader`
* `AutoDiscoveredAndroidInstrumentationLoader`

My preference is for the last one. It's more verbose, but I hope it's very clear in what it does.

Reply:
I'm always looking for name ideas, as that's the part I struggle the most with, so thanks for your suggestions!

I went with `AndroidInstrumentationLoader` because I think it's the most concise. Since users might need to access this type to configure some instrumentations I think it's better to keep the name as short as possible.